### ISSUE TITLE
Rule "arrow-parens" conflicting with the async keyword

### ISSUE THREAD

**Issue by buzinas** [eslint/eslint#4346]
#### 1.  The version of ESLint you are using
v1.8.0

#### 2. What you did
I wrote an async arrow function:

 ```javascript
 app
   .use(async (ctx, next) => {
     const start = new Date();
     await next();
     const ms = new Date() - start;
     console.log(`${ctx.method} ${ctx.url} - in ${ms}ms`);
   });
 ```
  And I have the "arrow-parens" rule set:
 ```json
 "arrow-parens": [2, "as-needed"]
 ```
#### 3. What you expected to happen
I expected that the rule should pass with no problems

#### 4. What actually happened
ESLint is throwing an error:
 ```console
 error  Expected parentheses around arrow function argument  arrow-parens

 ✖ 1 problem (1 error, 0 warnings)

 ```

**Comment by ilyavolodin** [comment]
This is most likely due to `async`. ESLint rules do not support experimental features. So while `babel-eslint` can parse and apply rules to your code, we can't really guarantee that the rules will behave correctly. `Babel-eslint` has a plugin that overrides some rules, it might be worth creating an issue there.
Closing, since this is not supported. Please feel free to reopen if you can reproduce this behavior without using experimental features.

**Comment by buzinas** [comment]
Maybe if someone else ends up here, I've just found this repo:
https://github.com/babel/eslint-plugin-babel

It creates some 1-to-1 rules of some unsupported ESLint rules as the one I reported here.
